### DISCUSSION TITLE
Scaling study/Profiling tools

### DISCUSSION THREAD

**Discussion by kyriakosTapinou** [PrincetonUniversity/athena#590] [Q&A]
I was wondering if there are any inbuilt diagnostics or metrics that can be output to measure performance, I'm looking at performance on my cluster. Is the "cpu-time-used" in the default output the total run time (serial+parallel run time)?

**Comment by tomidakn** [comment]
Athena++ outputs zone-cycles/cpu_second at the end of a simulation, which is exactly what you need.

**Comment by c-white** [comment]
And the cpu time is the walltime of the simulation, excluding a bit of setup in the beginning, independent of what things are computed in parallel. It also excludes the time it takes your cluster to acquire and free resources, which can be long relative to a short test and would be included in the scheduler's report of how long you ran.
